modules/shared/src/main/scala/io/constellationnetwork/currency/swap/ConsensusInput.scala

```diff
@@ -0,0 +1,60 @@
+package io.constellationnetwork.currency.swap
+
+import cats.Show
+
+import io.constellationnetwork.schema.peer.PeerId
+import io.constellationnetwork.schema.round.RoundId
+import io.constellationnetwork.schema.swap.SwapTransaction
+import io.constellationnetwork.security.Encodable
+import io.constellationnetwork.security.signature.Signed
+import io.constellationnetwork.security.signature.signature.Signature
+
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+
+sealed trait ConsensusInput
+
+object ConsensusInput {
+
+  sealed trait OwnerConsensusInput extends ConsensusInput
+  case object OwnRoundTrigger extends OwnerConsensusInput
+  case object InspectionTrigger extends OwnerConsensusInput
+
+  @derive(encoder, decoder)
+  sealed trait PeerConsensusInput extends ConsensusInput {
+    val senderId: PeerId
+    val owner: PeerId
+  }
+
+  @derive(encoder, decoder)
+  case class Proposal(
+    roundId: RoundId,
+    senderId: PeerId,
+    owner: PeerId,
+    facilitators: Set[PeerId],
+    transactions: Set[Signed[SwapTransaction]]
```

Comment:
is the swap transaction signed by the proposing peer?

Reply:
I don't know who should sign it. @ryle-goehausen? Here, I just need a proof that the transaction is correctly signed.

Reply:
I think this is alright given that the only swap transaction is AllowSpend, which is user generated. It is the balance update request that the AMM generates that has to be treated as rewards.